### DISCUSSION TITLE
How can I find out that the bot is blocked?

### DISCUSSION THREAD

**Discussion by AlexFox27SuperMan** [westacks/telebot#75] [Q&A] [answered]
For the second day I've been scratching my head, can I find out that the bot is blocked?
I wrote such a code, when I see it, that is, if the message does not reach, then the bot is blocked, but I do not fall into the error block
```php
 public function toTelegram($notifiable)
    {
        $chatId = $this->chatId;
        $message = $this->message;
        $bot = new TelegramNotification();

        try {
            $response = $bot->sendMessage([
                'chat_id' => $chatId,
                'text' => $message,
                'parse_mode' => 'HTML',
                'exceptions' => true,
            ]);


            return $response;
        } catch (TeleBotException $e) {
            if ($e->getCode() == 403 && strpos($e->getMessage(), 'bot was blocked by the user') !== false) {
                // Действия при блокировке бота пользователем
                Log::error('Бот заблокирован пользователем.');
            } else {
                // Другие действия при других ошибках Telegram
                Log::error('Ошибка отправки сообщения в Telegram: ' . $e->getMessage());
            }
        } catch (Exception $e) {
            // Действия при других исключениях
            Log::error('Ошибка отправки сообщения: ' . $e->getMessage());
        }
    }
```

**Comment by punyflash** [comment, marked as answer]
The reason you don't get an exception there is because `TelegramNotification` class is only creating a dataset for future notification, so there are no errors yet. Sending of notification happens internally https://github.com/westacks/telebot/blob/c00ad32a35e0b5064ac343b08db601865852e283/src/Laravel/Notifications/TelegramChannel.php#L27-L58

If you want to handle some logic if user notification failed - you need to create [event listener](https://laravel.com/docs/10.x/events) for `Illuminate\Notifications\Events\NotificationFailed` event
